core/src/main/java/gov/nist/secauto/metaschema/core/metapath/function/library/DefaultFunctionLibrary.java

```diff
@@ -146,10 +146,11 @@ public DefaultFunctionLibrary() { // NOPMD - intentional
     // https://www.w3.org/TR/xpath-functions-31/#func-static-base-uri
     registerFunction(FnStaticBaseUri.SIGNATURE);
     // P0: https://www.w3.org/TR/xpath-functions-31/#func-string
+    registerFunction(FnSubstring.SIGNATURE_THREE_ARG);
```

Comment:
@david-waltermire, having reviewed your other PR, for optional arguments I should be expressly declaring different arity functions separately? I don't remember doing that before and somehow dropping optional arguments in code, so I assume I should be adding this function registry change?

```suggestion
    registerFunction(FnSubstring.SIGNATURE_TWO_ARG);
    registerFunction(FnSubstring.SIGNATURE_THREE_ARG);
```

If so I'll adjust the other parts of the code, but I had not noticed this pattern before, so I thought I'd just ask.

Reply:
Specifying each variant is important to support the type coercion needed to support the function call. This happens automatically by the code that calls the function.